### PR TITLE
Add support for where_object and where_object_changes with JSON and JSONB columns

### PR DESCRIPTION
**Pull request by mikeraimondi**

### CODE REVIEW

**Comment by mikeraimondi** [comment]
The tests pass for me locally. Not sure what's going on with Travis.

File: lib/paper_trail/version_concern.rb [hunk 1]
```diff
@@ -118,12 +139,12 @@ def primary_key_is_int?
 
       # Returns whether the `object` column is using the `json` type supported by PostgreSQL
       def object_col_is_json?
-        @object_col_is_json ||= [:json, :jsonb].include?(columns_hash['object'].type)
```
Comment (batter) [inline lib/paper_trail/version_concern.rb]: Any reason we shouldn't cache this?

**Comment by batter** [comment]
Hmm.. I just upgraded to Postgres 9.4 on my local system and ran the test suite and saw the same errors that Travis is returning so I'll see if I can do some hacking to figure out what's up. Thanks for the pull request, let me know if you have any ideas regarding the invalid token errors.

**Comment by batter** [comment]
Thanks a lot for the pull request! This is super helpful. I was able to get all the tests to pass on Travis and even added some additional tests for good measure surrounding the column types.

It turns out that `ActiveRecord` 3.x doesn't support `JSON` column types for PostgreSQL, and `JSONB` column types are only supported on `ActiveRecord` 4.2+.

**Comment by mikeraimondi** [comment]
Nice! Glad you got it to pass on Travis.

**Comment by gmhawash** [comment]
Any plans on merging this PR into master?

**Comment by jaredbeck** [comment]
> Any plans on merging this PR into master?

The commits are already in master.  See, e.g. https://github.com/airblade/paper_trail/commit/a98d88cb309707bab2790550ce6c56c08185822b

GH doesn't make it easy to tell they're already in master, does it :)

**Comment by gmhawash** [comment]
Thanks Jared, I did not bother to look in the actual file, and you're right, GH did not indicate that it was merged in

**Pull request closed without merging**